### PR TITLE
DPP-432 Add exception tests to the JdbcLedgerDao suite

### PR DESCRIPTION
**Pull request by rautenrieth-da**
This PR adds coverage for Daml exceptions for the JdbcLedgerDao suite.

These tests overlap with the ledger API test tool, but only for ledgers that don't prune (or otherwise preprocess) Rollback nodes before the `ReadService` interface. By adding the test to the JdbcLedgerDao suite, we make sure exceptions are covered independent of the ledger implementations in this repository.

Additionally, one test uses a transaction that can only be produced by a privacy-aware ledger, which is something the ledger API test tool can not cover in this repository.

### CODE REVIEW

File: ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoExceptionSpec.scala [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    val create1 = createNode(createCid, stakeholders, stakeholders)
+    val create2 = createNode(fetcherCid, divulgees, stakeholders.union(divulgees))
+    val exercise1 = exerciseNode(fetcherCid).copy(
+      consuming = false,
+      actingParties = stakeholders,
+      signatories = divulgees,
+      stakeholders = stakeholders.union(divulgees),
+    )
+    val rollback = builder.rollback()
+    val exercise2 = fetchNode(createCid).copy(
```
Comment (cocreature) [inline ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoExceptionSpec.scala:131]: same comment here this seems like a fetch not an exercise.

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoExceptionSpec.scala [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      _ <- store(offsetAndEntry)
+      result1 <- contractsReader.lookupActiveContractAndLoadArgument(Set(alice), cid1)
+      result2 <- contractsReader.lookupActiveContractAndLoadArgument(Set(alice), cid2)
+    } yield {
+      result1 shouldBe None
+      result2.value shouldBe a[ContractInst[_]]
+    }
+  }
+
+  it should "divulge contracts fetched under rollback nodes" in {
```
[the remark this answers is not in the document]
Comment (rautenrieth-da) [reply ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoExceptionSpec.scala:46]: How can a contract be _disclosed_ through a child of a rollback node?

[the remark this answers is not in the document]
Comment (cocreature) [reply ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoExceptionSpec.scala:46]: that is a very good point. the create will clearly be rolled back. Sorry wasn’t thinking clearly :)

**Pull request merged**